samples_project/Assets/SampleViewer/Samples/FeatureLayer/FeatureLayer.cs

```diff
@@ -0,0 +1,407 @@
+    private FeatureLayerUIManager UIManager;
+
+    public List<Feature> Features = new List<Feature>();
+    public bool GetAllFeatures = true;
+    public bool GetAllOutfields = true;
+    public JToken[] jFeatures;
+    public int LastValue = 1;
+    public List<GameObject> FeatureItems = new List<GameObject>();
+    public List<Toggle> ListItems = new List<Toggle>();
+    public bool NewLink = true;
+    public GameObject OutfieldItem;
+    public List<string> OutfieldsToGet = new List<string>();
+    public int StartValue;
+    public WebLink WebLink;
+
+    private void Start()
+    {
+        CreateLink(WebLink.Link);
+        StartCoroutine(GetFeatures());
+        UIManager = GetComponent<FeatureLayerUIManager>();
+    }
+
+    public void CreateLink(string link)
+    {
+        EmptyOutfieldsDropdown();
+        if (link != null)
+        {
+            foreach (var header in WebLink.RequestHeaders)
+            {
+                if (!link.ToLower().Contains(header))
+                {
+                    link += header;
+                }
+            }
+
+            WebLink.Link = link;
+        }
+    }
+
+    public IEnumerator GetFeatures()
+    {
+        // To learn more about the Feature Layer rest API and all the things that are possible checkout
+        // https://developers.arcgis.com/rest/services-reference/enterprise/query-feature-service-layer-.htm
+
+        UnityWebRequest Request = UnityWebRequest.Get(WebLink.Link);
+        yield return Request.SendWebRequest();
+
+        if (Request.result != UnityWebRequest.Result.Success)
+        {
+            Debug.Log(Request.error);
+            UIManager.DisplayText = FeatureLayerUIManager.TextToDisplay.LinkError;
+        }
+        else
+        {
+            if (NewLink)
+            {
+                PopulateOutfieldsDropdown(Request.downloadHandler.text);
+            }
+            else
+            {
+                CreateGameObjectsFromResponse(Request.downloadHandler.text);
+                MoveCamera();
+                if (UIManager.DisplayText != FeatureLayerUIManager.TextToDisplay.CoordinatesError
+                    && UIManager.DisplayText != FeatureLayerUIManager.TextToDisplay.IndexOutOfBoundsError)
+                {
+                    UIManager.DisplayText = FeatureLayerUIManager.TextToDisplay.Information;
+                }
+            }
+            
+            NewLink = false;
+        }
+    }
+
+    private void CreateGameObjectsFromResponse(string Response)
+    {
+        // Deserialize the JSON response from the query.
+        var jObject = JObject.Parse(Response);
+        jFeatures = jObject.SelectToken("features").ToArray();
+
+        if (jFeatures[0].SelectToken("geometry").SelectToken("type").ToString().ToLower() == "point")
+        {
+            if (GetAllFeatures)
+            {
+                foreach (var feature in jFeatures)
+                {
+                    Feature currentFeature = new Feature();
```

Comment:
Each of the code chunks on ln 134, ln 193, etc, feel like they could be their own functions so you're not repeating code and having this massive function